sql/src/test/java/org/apache/druid/quidem/DruidQuidemCommandHandler.java

```diff
@@ -245,6 +269,159 @@ static class DruidPlanCommand extends AbstractRelPlanCommand
     }
   }
 
+  static class HintPlanCommand extends AbstractRelPlanCommand
+  {
+    HintPlanCommand(List<String> lines, List<String> content)
+    {
+      super(lines, content, DruidHook.DRUID_PLAN);
+    }
+
+    @Override
+    protected String getString(RelNode node)
+    {
+      final List<String> hintsCollect = new ArrayList<>();
+      final HintCollector collector = new HintCollector(hintsCollect);
+      node.accept(collector);
+      StringBuilder builder = new StringBuilder();
+      for (String hintLine : hintsCollect) {
+        builder.append(hintLine).append("\n");
+      }
+
+      return builder.toString();
+    }
+
+    private static class HintCollector extends RelShuttleImpl
```

Comment:
note: instead of implementing all these methods; you could subclass `RelHomogeneousShuttle` and implemented it only for `RelNode`